samples/net/sockets/echo_server/boards/atsamr21_xpro.conf

```diff
@@ -1,23 +1,31 @@
 #
-# Copyright (c) 2020-2021, Gerson Fernando Budke
 # Copyright (c) 2019, Benjamin Valentin
+# Copyright (c) 2020-2023, Gerson Fernando Budke
 #
 # SPDX-License-Identifier: Apache-2.0
 #
 
-# Reduced buffers to fit into SAMR21 SoC
+# Reduced configs to fit into SAMR21 SoC
 CONFIG_CPP=n
 
-CONFIG_NET_PKT_RX_COUNT=6
-CONFIG_NET_PKT_TX_COUNT=6
-CONFIG_NET_BUF_RX_COUNT=6
-CONFIG_NET_BUF_TX_COUNT=6
-CONFIG_NET_MAX_CONTEXTS=4
+CONFIG_NET_PKT_RX_COUNT=5
+CONFIG_NET_PKT_TX_COUNT=5
+CONFIG_NET_BUF_RX_COUNT=5
+CONFIG_NET_BUF_TX_COUNT=5
+CONFIG_NET_MAX_CONTEXTS=2
 CONFIG_NET_MAX_CONN=1
 CONFIG_NET_MAX_ROUTES=1
 CONFIG_NET_MAX_NEXTHOPS=1
 
-CONFIG_SHELL_STACK_SIZE=768
-CONFIG_SHELL_CMD_BUFF_SIZE=64
+CONFIG_SHELL_STACK_SIZE=384
+CONFIG_SHELL_CMD_BUFF_SIZE=16
 CONFIG_SHELL_ARGC_MAX=6
-CONFIG_SHELL_HISTORY_BUFFER=64
+CONFIG_SHELL_HISTORY_BUFFER=4
+
```

Comment:
Following things could be disabled without too much problems:
```
CONFIG_NET_SHELL_DYN_CMD_COMPLETION=n
CONFIG_NET_INTERFACE_NAME=n
CONFIG_NET_CONTEXT_REUSEADDR=n
CONFIG_NET_CONTEXT_REUSEPORT=n
```
The net-shell dynamic cmd completion is nice helper but can be safely turned off.
The network interface name was just introduced but for echo-* apps it is not needed.
Depending on sample app needs, the two reuse options are optional and can be disabled.

Reply:
Applied NET_CONTEXT_REUSE